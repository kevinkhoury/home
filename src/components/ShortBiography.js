import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'
import { addTrailingSlash } from '../utils'
import PreviewableImage from './PreviewableImage'
import { featuredImagePropTypes } from '../proptypes'

const ShortBiography = ({
  learnMoreButton,
  shortBiography,
  image,
  isPreview,
}) => (
  <div className="row" style={{ alignItems: 'center' }}>
    <div className="col-6">
      {!!shortBiography && <p>{shortBiography}</p>}
      {!!learnMoreButton && (
        <Link className="button" to={addTrailingSlash(learnMoreButton.link)}>
          {learnMoreButton.label}
        </Link>
      )}
    </div>
    <div className="col-6">
      {!!image && !!image.src && !!image.alt && (
        <figure className="gatsby-resp-image-card-no-margin">
          <PreviewableImage
            isPreview={isPreview}
            src={isPreview ? image.src : image.square}
            alt={image.alt}
            caption={image.caption}
          />
        </figure>
      )}
    </div>
  </div>
)

ShortBiography.propTypes = {
  learnMoreButton: PropTypes.shape({
    link: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
  }),
  shortBiography: PropTypes.string,
  image: featuredImagePropTypes,
  isPreview: PropTypes.bool,
}

export default ShortBiography
