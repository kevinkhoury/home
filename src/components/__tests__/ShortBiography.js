import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import ShortBiography from '../ShortBiography'

const DEFAULT_PROPS = {
  learnMoreButton: {
    link: 'link',
    label: 'label',
  },
  shortBiography: 'shortBiography',
  image: {
    src: {
      childImageSharp: {
        fluid: {},
        original: {
          height: 100,
          width: 100,
        },
      },
    },
    m: {
      childImageSharp: {},
    },
    d: {
      childImageSharp: {},
    },
    square: {
      childImageSharp: {},
    },
    alt: 'alttext',
    caption: 'captiontext',
  },
  isPreview: false,
}

jest.mock('./PreviewableImage', () => () => 'previewableImage')

afterEach(() => {
  jest.clearAllMocks()
})

describe('ShortBiography', () => {
  it('snapshot renders properly', () => {
    const tree = renderer.create(<ShortBiography {...DEFAULT_PROPS} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
  it('renders empty divs if no content provided', () => {
    render(<ShortBiography />)
    expect(screen.queryByText('label')).toBeNull()
    expect(screen.queryByText('shortBiography')).toBeNull()
    expect(screen.queryByAltText('alttext')).toBeNull()
    expect(screen.queryByText('captiontext')).toBeNull()
  })
})
