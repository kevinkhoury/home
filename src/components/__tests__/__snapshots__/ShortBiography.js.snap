// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortBiography snapshot renders properly 1`] = `
<div
  className="row"
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <div
    className="col-6"
  >
    <p>
      shortBiography
    </p>
    <a
      className="button"
      href="/link/"
    >
      label
    </a>
  </div>
  <div
    className="col-6"
  >
    <figure
      className="gatsby-resp-image-card-no-margin"
    >
      <img
        alt="alttext"
        className="gatsby-resp-image-image"
        src={
          Object {
            "childImageSharp": Object {},
          }
        }
      />
      <figcaption>
        captiontext
      </figcaption>
    </figure>
  </div>
</div>
`;
